docs: release notes (#3860)
scalameta/metals: website/blog/2022-04-25-aluminium.md

  </tr>
  <tr>
    <td>New features</td>
    <td align="center">7</td>
  </tr>
</tbody>
</table>

This release uses a new version of [Bloop](https://github.com/scalacenter/bloop/blob/main/notes/v1.5.0.md). 
This should fix a few bugs regarding e.g. stale diagnostics in Scala 3. Moreover, the new release brings also 
a few UX improvements regarding Doctor view and running/debugging your code.

For full details: https://github.com/scalameta/metals/milestone/48?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs and
Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- Add more information to the Doctor view
- Create companion object code action
- Add status bar when starting a debug session
- Ensure the "no run or test" message is shown to user.
- Show better unnaply signatures
- Provide an easier way to configure bloop settings
- Better MUnit support in Test Explorer

> 

Reviewer: ```suggestion
- Better sbt BSP integration
- Improved Scala3 support

```

a few UX improvements regarding Doctor view and running/debugging your code.

For full details: https://github.com/scalameta/metals/milestone/48?closed=1

Metals is a language server for Scala that works with VS Code, Vim, Emacs and
Sublime Text. Metals is developed at the
[Scala Center](https://scala.epfl.ch/) and [VirtusLab](https://virtuslab.com)
with the help from [Lunatech](https://lunatech.com) along with contributors from
the community.

## TL;DR

Check out [https://scalameta.org/metals/](https://scalameta.org/metals/), and
give Metals a try!

- Add more information to the Doctor view
- Create companion object code action
- Add status bar when starting a debug session
- Ensure the "no run or test" message is shown to user.
- Show better unnaply signatures
- Provide an easier way to configure bloop settings
- Better MUnit support in Test Explorer
- Better sbt BSP integration
- Improved Scala3 support

## Add more information to the Doctor view

Thanks to the joined effort of [\#3772](https://github.com/scalameta/metals/pull/3772), [\#3763](https://github.com/scalameta/metals/pull/3763) and [\#3710](https://github.com/scalameta/metals/pull/3710) the Doctor view currently:

> - shows Metals server version and information about Java used to run Metals server

Reviewer 2: ```suggestion
- shows the Metals server version and information about Java used to run the Metals server
```

> - displays compilation status of project

Reviewer 2: ```suggestion
- displays the compilation status of project
```

- allows to navigate to the build target info for each target

![new-doctor-view](https://imgur.com/ByzzlM8.png)

## Create companion object code action

Thanks to the efforts of [zmerr](https://github.com/zmerr), Metals offers a new code action - `Create companion object`.
As its name suggests, it can be used to automatically create a companion object for the given class, trait or enum.

![create-companion-object-code-action](https://imgur.com/HW5rr5f.gif)
Theme: [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme)

## Add status bar when starting a debug session

Starting a debug session is a complex task, which requires a few steps:

- Metals has to compile your project and create a debug session
- debugger has to be initialized (configuration, breakpoints, etc.)

> Very often each of these steps needs even a few seconds in order to complete.

Reviewer 2: ```suggestion
Very often each of these steps takes up to a few seconds in order to complete.
```

Until now, the user had no idea what is happening because Metals didn't show any progress indicator.
[ckipp01](https://github.com/ckipp01) addressed this problem and added a status bar for each of the steps.

![start-debug-session-status-bar](https://imgur.com/qh7Jzdy.gif)
Theme: [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme)

## Ensure the "no run or test" message is shown to user.

There is a command `Run main class or tests in current file` which is a convenient way, as name suggests, of running main class:
![run-main](https://imgur.com/rShZ8L3.gif)
Theme: [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme)

or running tests in the current file:
![run-tests](https://imgur.com/BtbOubC.gif)
Theme: [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme)

> It is also possible to just press `F5` in file and Metals will execute main class or run tests in the current file.

Reviewer 2: ```suggestion
It is also possible to just press `F5` in file and Metals will execute main class or run tests in the current file if no run configuration is defined.
```

configured through a global config file. Thanks to the [zmerr](https://github.com/zmerr), Metals is capable of changing those settings.
Each modification of `Bloop Jvm Properties` settings will try to update that global file if possible.
In case of manually configured settings, Metals will inform user about it and ask for their action.

![update-bloop-settings](https://imgur.com/Xz2gO0h.gif)
Theme: [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme)

## Better MUnit support in Test Explorer

Disclaimer: **this works only when Bloop is a build server**

Test Explorer in Metals is now able to detect some single test cases for [MUnit](https://scalameta.org/munit/) test framework.
![munit-single-tests](https://imgur.com/QQMLy6M.png)
Theme: [One Dark Pro](https://marketplace.visualstudio.com/items?itemName=zhuangtongfa.Material-theme)

More information about current MUnit support status can be found at this [issue](https://github.com/scalameta/metals/issues/3771).

## Better sbt BSP integration

Metals now doesn't run compilation when sbt generates files under the `src-managed` directory.

> This improvement fixed the continuous compilation when using sbt BSP and some sbt-plugins generate files on compile.

Reviewer 2: ```suggestion
The previous behaviour was related to some of the source generating sbt plugins. When used with sbt BSP users would experience  continuous compilation.
```
